File: 06-04-2024/Count ways to N'th Stair.js
//{ Driver Code Starts
// Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    for (let i = 0; i < t; i++) {
        let input_line = readLine().split(' ');
        let n = parseInt(input_line[0]);

        let obj = new Solution();
        let ans = obj.countWays(n);
        if (ans == -0n) ans = 0n;
        console.log(ans.toString());
    }
}

// } Driver Code Ends




// User function Template for javascript

/**
 * @param {number} m
 * @returns {BigInt}
 */

class Solution {
    // Function to count number of ways to reach the nth stair
    // when order does not matter.
    countWays(n) {
      //Your Code Start
         return Math.floor(n/2+1);
      //Your Code End
    }
}

File: 06-04-2024/Count ways to N'th Stair.cs
//{ Driver Code Starts
// INITIAL CODE
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode {

class GFG {
    static void Main(string[] args) {
        int testcases; // Taking testcase as input
        testcases = Convert.ToInt32(Console.ReadLine());
        while (testcases-- > 0) // Looping through all testcases
        {
            int N = Convert.ToInt32(Console.ReadLine());
            Solution obj = new Solution();
            long res = obj.countWays(N);

            Console.Write(res + "\n");
        }
    }
}
}

// } Driver Code Ends




// User function Template for C#
class Solution {
    // Complete this function
    // Function to count number of ways to reach the nth stair
    // when order does not matter.
    public long countWays(int n) {
        // Your Code Start
         return n/2+1;
      //Your Code End
    }
}
